src/main/java/org/jabref/gui/maintable/MainTableHeaderContextMenu.java

```diff
@@ -0,0 +1,184 @@
+package org.jabref.gui.maintable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javafx.scene.control.ContextMenu;
+import javafx.scene.control.RadioMenuItem;
+import javafx.scene.control.SeparatorMenuItem;
+import javafx.scene.control.TableColumn;
+import javafx.scene.layout.StackPane;
+
+import org.jabref.gui.maintable.columns.MainTableColumn;
+
+public class MainTableHeaderContextMenu extends ContextMenu {
+
+    MainTable mainTable;
+    MainTableColumnFactory factory;
+
+    /**
+     * Constructor for the right click menu
+     *
+     */
+    public MainTableHeaderContextMenu(MainTable mainTable, MainTableColumnFactory factory) {
+        super();
+        this.mainTable = mainTable;
+        this.factory = factory;
+        constructItems(mainTable);
+    }
+
+    /**
+     * Handles showing the menu in the cursors position when right-clicked.
+     */
+    public void show(boolean show) {
+        // TODO: 20/10/2022 unknown bug where issue does not show unless parameter is passed through this method.
+        mainTable.setOnContextMenuRequested(event -> {
+            // Display the menu if header is clicked, otherwise, remove from display.
+            if (!(event.getTarget() instanceof StackPane) && show) {
+                this.show(mainTable, event.getScreenX(), event.getScreenY());
+            } else if (this.isShowing()) {
+                this.hide();
+            }
+            event.consume();
+        });
+    }
+
+    /**
+     * Constructs the items for the list and places them in the menu.
+     */
+    private void constructItems(MainTable mainTable) {
+        // Reset the right-click menu
+        this.getItems().clear();
+
+        // Populate the menu with the commonly used fields
+        for (TableColumn<BibEntryTableViewModel, ?> tableColumn:commonColumns()) {
```

Comment:
If the user reorders the columns differently than what is in "commonColumns", then it might get confusing since the column order then differs from the order in the context menu.

For this reason, I would swap the logic here: first display all currently shown columns in the display order, and then after the separator list common column that are not yet displayed.

This also plays well with the "addColumn" function, which adds the columns at the end.

Reply:
Switched this logic around :)